packages/blade/src/components/Card/CardHeader.tsx

```diff
@@ -0,0 +1,194 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React from 'react';
+import { Divider } from './Divider';
+import { useInsideCard } from './CardContext';
+import Box from '~components/Box';
+import type { TextProps, TextVariant } from '~components/Typography';
+import { Heading, Text } from '~components/Typography';
+import type { IconComponent } from '~components/Icons';
+import type { CounterProps } from '~components/Counter';
+import { Counter } from '~components/Counter';
+import type { BadgeProps } from '~components/Badge';
+import { Badge } from '~components/Badge';
+import type { LinkProps } from '~components/Link';
+import { Link } from '~components/Link';
+import type { ButtonProps } from '~components/Button';
+import { Button } from '~components/Button';
+import { minHeight } from '~components/Button/BaseButton/buttonTokens';
+import { makeSpace } from '~utils';
+
+type CardHeaderProps = {
+  children?: React.ReactNode;
+};
+
+const Components = {
+  CardHeaderIcon: 'CardHeaderIcon',
+  CardHeaderCounter: 'CardHeaderCounter',
+  CardHeaderBadge: 'CardHeaderBadge',
+  CardHeaderText: 'CardHeaderText',
+  CardHeaderLink: 'CardHeaderLink',
+  CardHeaderIconButton: 'CardHeaderIconButton',
+};
+
+const CardHeaderIcon = ({ icon: Icon }: { icon: IconComponent }): React.ReactElement => {
+  useInsideCard('CardHeaderIcon');
+
+  return <Icon color="surface.text.normal.lowContrast" size="xlarge" />;
+};
+CardHeaderIcon.componentId = Components.CardHeaderIcon;
+
+const CardHeaderCounter = (props: CounterProps): React.ReactElement => {
+  useInsideCard('CardHeaderCounter');
+
+  return <Counter {...props} />;
+};
+CardHeaderCounter.componentId = Components.CardHeaderCounter;
+
+const CardHeaderBadge = (props: BadgeProps): React.ReactElement => {
+  useInsideCard('CardHeaderBadge');
+
+  return <Badge {...props} />;
+};
+CardHeaderBadge.componentId = Components.CardHeaderBadge;
+
+const CardHeaderText = (props: TextProps<{ variant: TextVariant }>): React.ReactElement => {
+  useInsideCard('CardHeaderText');
+
+  return <Text {...props} />;
+};
+CardHeaderText.componentId = Components.CardHeaderText;
+
+const CardHeaderLink = (props: LinkProps): React.ReactElement => {
+  useInsideCard('CardHeaderLink');
+
+  return <Link {...props} />;
+};
+CardHeaderLink.componentId = Components.CardHeaderLink;
+
+type CardHeaderIconButtonProps = Omit<
+  ButtonProps,
+  'variant' | 'size' | 'iconPosition' | 'isFullWidth' | 'children'
+> & {
+  icon: IconComponent;
+};
+
+const CardHeaderIconButton = (props: CardHeaderIconButtonProps): React.ReactElement => {
+  useInsideCard('CardHeaderIconButton');
+
+  return (
+    <Box width={makeSpace(minHeight.xsmall)}>
```

Comment:
To control the width of the button & make the size 1:1. 

Otherwise the width get wider: 

<img width="75" alt="image" src="https://user-images.githubusercontent.com/35374649/208101106-6e7ceb5e-c7c7-4e25-a085-4f53b04a808b.png">

With box: 

<img width="66" alt="image" src="https://user-images.githubusercontent.com/35374649/208101169-a201fb40-98cc-429a-b7d9-59db9869110e.png">

Reply:
which is fine? because that is what generally happens with our regular icon buttons?

Reply:
But in design the card's icon button width:height is 1:1 which we need to follow I think, cc @saurav12 ^

Reply:
then we shall add this to our icon button base component directly since it should look same across